### ISSUE TITLE
pkg_system_requirements only reports direct sys reqs and ignores imports

### ISSUE THREAD

**Issue by assignUser** [r-lib/pak#315]
As I noticed in https://github.com/r-lib/actions/issues/370 `local_system_requirments` and `pkg_system_requirements` seem to behave unxepectedly different. `local` recursively reports all system requirements of the package and its dependencies, while `pkg` only reports requirements listed directly in the `DESCRIPTION` of the package. I assume this is a bug?

``` r
pth <- fs::path_temp("testpackage")
usethis::create_package(pth, rstudio = FALSE, check_name = FALSE, fields = list(Imports = "gert"))
#> v Creating ...
setwd(pth)
pak::local_install()
#> i Loading metadata database
#> i Loading metadata database
pak::local_system_requirements(os = "ubuntu", os_release = "18.04")
#> [1] "apt-get install -y software-properties-common"
#> [2] "add-apt-repository -y ppa:cran/libgit2"       
#> [3] "apt-get update"                               
#> [4] "apt-get install -y libcurl4-openssl-dev"      
#> [5] "apt-get install -y libssl-dev"                
#> [6] "apt-get install -y git"                       
#> [7] "apt-get install -y libgit2-dev"
pak::pkg_system_requirements("testpackage", os = "ubuntu", os_release = "18.04")
#> character(0)
```
<sup>Created on 2021-09-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by assignUser** [comment]
This seems to also cause `pak::pkg_install` to fail installation of e.g. devtools when dependencies have additional system requirements (in this case curl/openssl)

```r
> pak::pkg_install("devtools")
                                                                           
→ Will install 38 packages.
→ Will update 12 packages.
.
.
.
ℹ Building curl 4.3.2                             
✖ Failed to build curl 4.3.2                      
Error: Failed to build source package 'curl'    
```

**Comment by riccardoporreca** [comment]
> I still think that the different behavior should be noted in the docs and `pkg_system_requirements()` should communicate API failures to the user to make troubleshooting easier.

@assignUser, see commits above, merged by PR #328

**Comment by assignUser** [comment]
@riccardoporreca Great, thank you!
